server/opendp_apps/profiler/profile_runner.py

```diff
@@ -1,37 +1,222 @@
+"""
+Read and Profile a File.
+"""
+import os
+from django.db.models.fields.files import FieldFile
+
 from opendp_apps.analysis.models import DepositorSetupInfo
 from opendp_apps.dataset.models import DataSetInfo
+
+from opendp_apps.model_helpers.basic_err_check import BasicErrCheck
+
 from opendp_apps.profiler.dataset_info_updater import DataSetInfoUpdater
+from opendp_apps.profiler.csv_reader import CsvReader
 from opendp_apps.profiler.variable_info import VariableInfoHandler
+from opendp_apps.profiler import static_vals as pstatic
 
 
-def run_profile(df, dataset_info_object_id):
-    """
-    Process dataframe for variable profiling, while updating the DatasetInfo object at each step
-    :param df:
-    :param dataset_info_object_id:
-    :return: VariableInfoHandler
-    """
-    dataset_info = None
-    dataset_info_updater = None
+class ProfileRunner(BasicErrCheck):
```

Comment:
Part of the reason I removed this class was to make the code easier to follow - why do we need to add this back, rather than put error handling in CsvReader?

Reply:
Agreed on the easier to follow part. The main function being used is the "run_profile_by_filefield"

- https://github.com/opendp/dpcreator/blob/develop/server/opendp_apps/profiler/tasks.py#L18

The main reason for the changes was to:

(a) make it easier to propagate error messages rather than having to try/catch multiple exceptions each place it was used. The error messages -- say profiling a non csv-readable file -- eventually go back via websockets. With the latest chance, they could end with a "dead end".

(b) bundle the CsvReader functionality and "run_profile" functionality together b/c in the near future (next week?), the Django FileField may use S3 or Azure storage and will no longer have a ".path" attribute. e.g. There will need to be extra functionality in the CsvReader -- or before the CsvReader -- to handle that.  

That went back to (a) and not wanting to change the interface to the overall CsvReader/Profile usage. Keeping it as:

```
# prunner is an instance of the ProfileRunner rather than the result of `run_profile`
prunner = run_profile_by_filefield(self.dataset_info.object_id,
                                           max_num_features=settings.PROFILER_COLUMN_LIMIT)

if prunner.has_error():
   # do something
else:
   # do something
```
It's a bit gray and definitely willing to re-simplify it after testing with S3

Reply:
- Also allowing for the other near-future option of not saving the number of rows/dataset size based on the user answers to questions